src/Components/WebAssembly/WebAssembly/src/Services/Logging/WebAssemblyLogger.cs

```diff
@@ -0,0 +1,134 @@
+        {
+            var loggers = Loggers;
+            if (loggers == null)
+            {
+                return;
+            }
+
+            List<Exception> exceptions = null;
+            for (var i = 0; i < loggers.Length; i++)
+            {
+                ref readonly var loggerInfo = ref loggers[i];
+                var logger = loggerInfo.Logger;
+                if (!logger.IsEnabled(logLevel))
+                {
+                    continue;
+                }
+
+                LoggerLog(logLevel, eventId, logger, exception, formatter, ref exceptions, state);
+            }
+
+            if (exceptions != null && exceptions.Count > 0)
+            {
+                ThrowLoggingError(exceptions);
+            }
+
+            static void LoggerLog(LogLevel logLevel, EventId eventId, ILogger logger, Exception exception, Func<TState, Exception, string> formatter, ref List<Exception> exceptions, in TState state)
+            {
+                try
+                {
+                    logger.Log(logLevel, eventId, state, exception, formatter);
+                }
+                catch (Exception ex)
+                {
+                    if (exceptions == null)
+                    {
+                        exceptions = new List<Exception>();
+                    }
+
+                    exceptions.Add(ex);
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            var loggers = Loggers;
+            if (loggers == null)
+            {
+                return false;
+            }
+
+            List<Exception> exceptions = null;
+            var i = 0;
+            for (; i < loggers.Length; i++)
+            {
+                ref readonly var loggerInfo = ref loggers[i];
+                var logger = loggerInfo.Logger;
+                if (!logger.IsEnabled(logLevel))
+                {
+                    continue;
+                }
+
+                if (LoggerIsEnabled(logLevel, loggerInfo.Logger, ref exceptions))
+                {
+                    break;
+                }
+            }
+
+            if (exceptions != null && exceptions.Count > 0)
+            {
+                ThrowLoggingError(exceptions);
+            }
+
+            return i < loggers.Length ? true : false;
+
+            static bool LoggerIsEnabled(LogLevel logLevel, ILogger logger, ref List<Exception> exceptions)
+            {
+                try
+                {
+                    if (logger.IsEnabled(logLevel))
+                    {
+                        return true;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    if (exceptions == null)
+                    {
+                        exceptions = new List<Exception>();
+                    }
+
+                    exceptions.Add(ex);
+                }
+
+                return false;
```

Comment:
This logic to aggregate exceptions looks super robust, though TBH I'm not certain what the use case for it is. If we didn't catch any of this, and just let the first exception be unhandled (since it's never a legit scenario to have loggers throw), would that be sufficient? Are there cases where it's necessary to collect more information than that?

Reply:
This is heavily inspired by the [Logger](https://github.com/dotnet/extensions/blob/master/src/Logging/Logging/src/Logger.cs) implementation in Microsoft.Extensions.Logging.

The intent behind this logic was to collect all the exceptions that arise from each logger that is registered and throw an aggregate exception for them. It is possible to bail and throw at the first exception thrown from a logger but I figured I would keep this logic because:

- It's more helpful to the developer to get all the exceptions from buggy loggers instead of getting them one-by-one.
- Wanted to follow existing experiences that devs would be used to. This didn't seem, outrageous enough to nuke.